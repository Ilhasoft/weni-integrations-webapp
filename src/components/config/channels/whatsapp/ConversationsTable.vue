<template>
  <div class="conversations-table-container">
    <unnnic-table :items="items">
      <template v-slot:header>
        <unnnic-table-row :headers="headers" />
      </template>

      <template v-slot:item="{ item }">
        <unnnic-table-row :headers="headers">
          <template v-slot:userMessages>
            <div :title="item.userMessages" class="break-text">
              {{ item.userMessages }}
            </div>
          </template>

          <template v-slot:templateMessages>
            <div :title="item.templateMessages" class="break-text">
              {{ item.templateMessages }}
            </div>
          </template>

          <template v-slot:total>
            <div :title="item.total" class="break-text">
              {{ item.total }}
            </div>
          </template>
        </unnnic-table-row>
      </template>
    </unnnic-table>
  </div>
</template>

<script>
  export default {
    name: 'ConversationsTable',
    props: {
      userMessages: {
        type: Number,
        default: 0,
      },
      templateMessages: {
        type: Number,
        default: 0,
      },
      total: {
        type: Number,
        default: 0,
      },
    },
    data() {
      return {
        headers: [
          {
            id: 'userMessages',
            text: this.$t('WhatsApp.config.conversations.conversations_count.table.user_messages'),
            flex: 1,
          },
          {
            id: 'templateMessages',
            text: this.$t(
              'WhatsApp.config.conversations.conversations_count.table.template_messages',
            ),
            flex: 1,
          },
          {
            id: 'total',
            text: this.$t('WhatsApp.config.conversations.conversations_count.table.total'),
            width: '55px',
          },
        ],
      };
    },
    computed: {
      items() {
        return [
          {
            userMessages: this.userMessages,
            templateMessages: this.templateMessages,
            total: this.total,
          },
        ];
      },
    },
  };
</script>

<style scoped lang="scss"></style>
