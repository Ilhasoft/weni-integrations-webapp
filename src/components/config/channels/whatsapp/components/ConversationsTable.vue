<template>
  <div class="conversations-table-container">
    <unnnic-table :items="items">
      <template v-slot:header>
        <unnnic-table-row :headers="headers" />
      </template>

      <template v-slot:item="{ item }">
        <unnnic-table-row :headers="headers">
          <template v-slot:userMessages>
            <div :title="item.userMessages" class="break-text">
              {{ item.userMessages }}
            </div>
          </template>

          <template v-slot:businessMessages>
            <div :title="item.businessMessages" class="break-text">
              {{ item.businessMessages }}
            </div>
          </template>

          <template v-slot:total>
            <div :title="item.total" class="break-text">
              {{ item.total }}
            </div>
          </template>
        </unnnic-table-row>
      </template>
    </unnnic-table>
  </div>
</template>

<script>
  export default {
    name: 'ConversationsTable',
    props: {
      userMessages: {
        type: Number,
        default: 0,
      },
      businessMessages: {
        type: Number,
        default: 0,
      },
      total: {
        type: Number,
        default: 0,
      },
    },
    data() {
      return {
        headers: [
          {
            id: 'userMessages',
            text: this.$t('WhatsApp.config.conversations.conversations_count.table.user_messages'),
            flex: 1,
          },
          {
            id: 'businessMessages',
            text: this.$t(
              'WhatsApp.config.conversations.conversations_count.table.business_messages',
            ),
            flex: 1,
          },
          {
            id: 'total',
            text: this.$t('WhatsApp.config.conversations.conversations_count.table.total'),
            width: '55px',
          },
        ],
      };
    },
    computed: {
      items() {
        return [
          {
            userMessages: this.userMessages,
            businessMessages: this.businessMessages,
            total: this.total,
          },
        ];
      },
    },
  };
</script>

<style scoped lang="scss">
  ::v-deep .scroll {
    padding-right: 0;
  }
</style>
