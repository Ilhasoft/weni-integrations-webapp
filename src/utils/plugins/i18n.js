import Vue from 'vue';
import VueI18n from 'vue-i18n';

Vue.use(VueI18n);

import en from '../../locales/en.json';
import pt_br from '../../locales/pt_br.json';
import es_es from '../../locales/es_es.json';

const languages = {
  'en-us': en,
  'pt-br': pt_br,
  es: es_es,
};

const messages = Object.assign(languages);

const i18n = new VueI18n({
  locale: 'en-us',
  fallbackLocale: 'en-us',
  messages,
});

export default i18n;
