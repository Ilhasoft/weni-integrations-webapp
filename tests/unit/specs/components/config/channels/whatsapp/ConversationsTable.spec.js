import VueRouter from 'vue-router';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import ConversationsTable from '@/components/config/channels/whatsapp/ConversationsTable.vue';
import i18n from '@/utils/plugins/i18n';

const localVue = createLocalVue();
localVue.use(VueRouter);
describe('whatsapp/ConversationsTable.vue', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(ConversationsTable, {
      localVue,
      i18n,
      propsData: {
        userMessages: 10,
        businessMessages: 20,
        total: 30,
      },
      mocks: {
        $t: () => 'some specific text',
      },
      stubs: {
        UnnnicTable: true,
        UnnnicTableRow: true,
      },
    });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should be rendered properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
