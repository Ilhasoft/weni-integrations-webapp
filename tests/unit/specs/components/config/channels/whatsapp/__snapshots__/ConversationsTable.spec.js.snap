// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`whatsapp/ConversationsTable.vue should be rendered properly 1`] = `
<div class="conversations-table-container">
  <unnnictable-stub items="[object Object]"></unnnictable-stub>
</div>
`;
